packages/roosterjs-editor-core/lib/coreApi/getSelectionRangeEx.ts

```diff
@@ -0,0 +1,75 @@
+import { EditorCore, GetSelectionRangeEx, SelectionRangeEx } from 'roosterjs-editor-types';
+import {
+    contains,
+    createRange,
+    NormalSelectionRange,
+    safeInstanceOf,
+    TableSelectionRange,
+    TableMetadata,
+    VTable,
+} from 'roosterjs-editor-dom';
+
+/**
+ * @internal
+ * Get current or cached selection range
+ * @param core The EditorCore object
+ * @returns A Range object of the selection range
+ */
+export const getSelectionRangeEx: GetSelectionRangeEx = (core: EditorCore) => {
+    let result: SelectionRangeEx = null;
+    if (core.lifecycle.shadowEditFragment) {
+        const shadowRange =
+            core.lifecycle.shadowEditSelectionPath &&
+            createRange(
+                core.contentDiv,
+                core.lifecycle.shadowEditSelectionPath.start,
+                core.lifecycle.shadowEditSelectionPath.end
+            );
+
+        if (shadowRange.collapsed) {
```

Comment:
why?

For shadow edit, we should not touch anything from the real DOM

Reply:
When the new API and Table Selection, the selection is always going to be collapsed, as it just stays in the first cell as we just change the styles of the cells. So I think we will need to check if the shadow range is collapsed, if there is a table selection. 

Otherwise if there is a better approach we would need to do it in the lifecycle plugin

Reply:
Better approach should be change the shadowEditSelectionPath, ... to make it knows table selection. But we can do this later.